nucypher/blockchain/eth/trackers/dkg.py

```diff
@@ -253,7 +261,10 @@ def scan(self):
         # Note that our chain reorg safety blocks cannot go negative
         if self.start_block is None:
             self.start_block = self._get_start_block_number()
-        start_block = max(self.state.get_last_scanned_block() - chain_reorg_safety_blocks, self.start_block)
+        start_block = min(
+            self.state.get_last_scanned_block() - chain_reorg_safety_blocks,
```

Comment:
Hmmm 🤔 ...what's the value of `get_last_scanned_block()` on startup?

`start_block` is only set once on startup, so using `min` here could cause it to stay at the same block forever, as the scan tries to progress over time. That's why `max` was used.

Reply:
Ok, that makes sense, but something's weird here. The behavior that prompted us to look here was that in the logs, we saw patterns were blocks seemed to be skipped like the following:
```
Scanning blocks: 1234 - 1234
... 
Scanning blocks: 1244 - 1244
...
Scanning blocks: 1255 - 1254   (yeah sometimes the start block was higher than the end block)
```
Also, the code in this function seem overly complex and trying to do many things at the same time.

Reply:
We also need some improved testing in this area.